providers/azure/azure.go

```diff
@@ -46,6 +46,9 @@ var DefaultConfig = Config{
 
 // Config Azure storage configuration.
 type Config struct {
+	ClientID                string             `yaml:"client_id"`
+	ClientSecret            string             `yaml:"client_secret"`
+	TenantID                string             `yaml:"tenant_id"`
```

Comment:
small nit: WDYT of changing the config field to one of these to avoid users possibly getting confused with `tenant_id` from either (Thanos/Loki/Mimir)?
```suggestion
	TenantID                string             `yaml:"az_tenant_id"`
```
or 

```suggestion
	TenantID                string             `yaml:"csc_tenant_id"`
```

Reply:
that's fair. I'm going for `az_tenant_id` as googling that term led to better results than `csc_tenant_id`